src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2874,41 +2874,20 @@ private void RemoveDirectoryInfoItem(DirectoryInfo directory, bool recurse, bool
                 continueRemoval = ShouldProcess(directory.FullName, action);
```

Comment:
It might be out the scope for this PR, but we don't check `continueRemoval` before removing the junction point ... that seems wrong.

Reply:
Let's postpone to new PR.

Reply:
I see a comment before the line says that we ask the confirmation only for removing root directory or recurse - but seems current behavior is that we remove the  junction point _as an object_ (not directory and no recurse) so we don't ask confirmation.